src/main/java/it/pagopa/pm/gateway/controller/PostePayPaymentTransactionsController.java

```diff
@@ -342,4 +362,100 @@ private String getCustomEnvironmentProperty(String parameterizedPropertyName, St
         }
 
     }
+
+    @DeleteMapping(REQUEST_PAYMENTS_POSTEPAY_REQUEST_ID)
+    public ResponseEntity<PostePayRefundResponse> refundPostePayPayment(@PathVariable String requestId,
+                                                                        @RequestHeader(required = false, value = MDC_FIELDS) String mdcFields) throws RestApiException {
+        setMdcFields(mdcFields);
+        log.info("START - refund PostePay request for paymentRequestEntity with guid: " + requestId);
+
+        PaymentRequestEntity requestEntity = paymentRequestRepository.findByGuid(requestId);
+
+        if (Objects.isNull(requestEntity) || !REQUEST_PAYMENTS_POSTEPAY.equals(requestEntity.getRequestEndpoint())) {
+            log.error("No PostePay response entity object has been found with guid " + requestId);
+            return createPostePayRefundResponse(requestId, null, null, ExceptionsEnum.TRANSACTION_NOT_FOUND);
+        }
+
+        if (requestEntity.getIsRefund()) {
+            log.info("Transaction with id " + requestEntity.getIdTransaction() + " has already be refunded");
+            return createPostePayRefundResponse(requestId, requestEntity.getCorrelationId(), null, ExceptionsEnum.TRANSACTION_ALREADY_REFUNDED);
+        }
+
+        // step 1. acquiring payment details PostePay side in order to validate the deletion request
+        DetailsPaymentRequest detailsPaymentRequest = createDetailPaymentRequest(requestEntity);
+
+        boolean isAuthCodeEmpty = StringUtils.isEmpty(requestEntity.getAuthorizationCode());
+        boolean checkDetail = true;
+
+        if (isAuthCodeEmpty) {
+            try {
+                log.info("Starting check Details for detailsPaymentRequest with guid: " + requestEntity.getGuid());
+                checkDetail = checkDetailStatus(detailsPaymentRequest);
+            } catch (Exception e) {
+                log.error("Generic Exception while checkDetailStatus DetailsPaymentRequest with paymentId: "
+                        + detailsPaymentRequest.getPaymentID() + " error message " + e.getMessage());
+            }
+        }
+
+        if (!isAuthCodeEmpty || checkDetail) {
```

Comment:
This condition/logic can be written differently:
- !isAuthcodeEmpty can be merged in an "else" clause

also, in the if statement above, maybe you can do something like:
isAuthCodeEmpty() ? --> checkDetailStatus. Status is ok? --> executeRefundRequest, else abort.
!isAuthCodeEmpty() (which could be an"else" branch): executeRefundRequest()